nmrfx-core/src/main/java/org/nmrfx/chemistry/io/PSFFile.java

```diff
@@ -68,126 +59,104 @@ public static void read(String fileName)
         int structureNumber = 0;
 
         Residue residue = null;
-        ArrayList<Atom> atoms = new ArrayList<Atom>();
+        ArrayList<Atom> atoms = new ArrayList<>();
         int natoms = 0;
         int nbonds = 0;
-
-        while (true) {
-            try {
+        try (BufferedReader bf = new BufferedReader(new FileReader(fileName));
+             LineNumberReader lineReader = new LineNumberReader(bf)) {
+            while (true) {
```

Comment:
One more 😄

Reply:
removed LineReader and replaced true with bufferReader.readLine